docs/designs/github-user-cache.md

```diff
@@ -45,3 +45,43 @@ A profile can be a valid GitHub user, an email that does not exist or a login th
 - *Clustered Build Server*:
     - Set `resolveGitHubUser` to `true` to resolve contributors.
     - Set `gitHubUserCache` to a shared blob storage to share it with multiple servers.
+
+## Cache life cycle
+
+Inspite of loading and saving cache locally, DocFX can upload it to remote server for other clients reuse in clustered build server scenario. ETag can be used to control upload conflict.
+
+### Download cache
+
+When restore a resource, DocFX can record the ETag infomation in filename for later use. The file name is of format, with `+` as separator:
+```
+{Sha1Hash(content)}+etag+UrlEncode(ETag)
```

Comment:
just another thought around this, instead of embedding etag and potentially other info into file name, what about a manifest json file that describes the download?
```json
{
    "https://github.com/a-url": {
        "sha1": "{content-sha1}",
        "etag": "etag",
        "last_modified_date": "last_modified_date"
    }
}
```

File layout:

```
/downloads/
      - downloads.json
      - {content-1-sha1}
      - {content-2-sha1}
      - ...
```

Reply:
I think the pros of a manifest file is that I can get modified date without enumerating directory. The cons is that this manifest need to be read/written concurrently if multiple DocFXs are running, and it introduces a dependency between manifest and cache, which cause extra complexity. Prefer the current design if only ETag is needed.

Reply:
if we introduce a manifest file, we need lock/unlock the file before modify it, which is what we want to avoid